python/taichi/lang/impl.py

```diff
@@ -72,6 +72,17 @@ def begin_frontend_struct_for(group, loop_range):
     taichi_lang_core.begin_frontend_struct_for(group, loop_range.ptr)
 
 
+def begin_frontend_if(cond):
+    if is_taichi_class(cond):
```

Comment:
A completely unrelated topic: I constantly get confused by `is_taichi_class` because i thought things like `Expr` would also belong to these category. Since this is only for matrix/vector, maybe we should explicitly name it `is_taichi_matrix` or `is_taichi_algebra_class` or something like that in the future?

Reply:
> Since this is only for matrix/vector,

Not really, FYI I have `ts.Complex` in [taichi_glsl](https://github.com/taichi-dev/taichi_glsl), `t3.Affine` in [taichi_three](https://github.com/taichi-dev/taichi_three), they're using the roughly same API as currently matrix have so `t3.Affine.field` would be as neat as `ti.Matrix.field` does, thanks to `is_taichi_class(x)` instead of just `isinstance(x, ti.Matrix)`.

> `is_taichi_algebra_class`

This name nit sounds better to me :)